### ISSUE TITLE
False positive for `arrow-body-style` rule when used as a noop

### ISSUE THREAD

**Issue by sindresorhus** [eslint/eslint#4411]
```
~/dev
❯ eslint --version                                                                            
v1.9.0
                                                                                                                                               
~/dev
❯ echo 'stream.on('error', () => {});' | eslint --stdin --env=es6 --rule='arrow-body-style: [2, "as-needed"]'

<text>
  1:24  error  Unexpected empty block in arrow body  arrow-body-style

✖ 1 problem (1 error, 0 warnings)
```

Sometimes you have to pass a noop function. In this example the braces are required, but ESLint still complains, even though the setting is `as-needed`.

**Comment by alberto** [comment]
From the docs:
> Additionally, this rule specifically warns against a possible developer error when the intention is to return an empty object literal but creates an empty block instead, returning undefined.

I think you'll have to use an inline comment to ignore that.

**Comment by sindresorhus** [comment]
I don't understand why that's mixed into this rule. It should at least be an option.

I think empty functions are much more common than returning an empty object.

**Comment by mysticatea** [comment]
I thought there are some notations for a noop arrow.

```js
let noop = () => {};
let noop = () => undefined;
let noop = () => void 0;
```

And `() => {}` is confusable to beginners IMO.
So I'd like to warn it by default.

I vote for an option, `[2, "as-needed", {"allowEmptyBlock": true}]` ?

**Comment by nzakas** [comment]
Yeah, I'm not sure why this check in this rule, either. The rule is about style, I think we should remove that extra check.

**Comment by mysticatea** [comment]
Hmm, as I mentioned above, there are several notations for a noop arrow, so I don't think the empty block in a arrow function's body is *needed*. So I think that to report the empty block by this rule is reasonable.

But I don't oppose strongly if separate that behavior to new rule such as `no-arrow-empty-block`.

**Comment by nzakas** [comment]
This rule is about style, which is just about using braces or not. For an empty body, you need the braces. We can argue that you can have other bodies for noop functions, but since this is just about style, you must have braces for a zero statement body.
